docs/source/onedal/svm/svm.rst

```diff
@@ -0,0 +1,188 @@
+         related computations (see [Joachims99]_).
+
+The solution of the problem defines the separating hyperplane and
+corresponding decision function :math:`D(x)= \sum_{k} {y_k \alpha_k K(x_k, x)} + b` 
+where only those :math:`x_k` that correspond to non-zero :math:`\alpha_k` appear in the sum,
+and :math:`b` is a bias. Each non-zero :math:`\alpha_k` is called a classification
+coefficient and the corresponding :math:`x_k` is called a support vector.
+
+.. _svm_t_math_smo:
+
+Training method: *smo*
+~~~~~~~~~~~~~~~~~~~~~~
+In *smo* training method, the all vectors from training dataset is used for each iteration.
+
+.. _svm_t_math_thunder:
+
+Training method: *thunder*
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+In *thunder* training method, the algorithm iteratively solves the convex optimization problem
+with the linear constraints by selecting the fixed set of active constrains (working set) and 
+applying Sequential Minimal Optimization (SMO) solver to the selected subproblem.
+The description of this method is given in Algorithm [Wen2018]_. 
+
+.. _svm_i_math:
+.. _svm_i_math_smo:
+.. _svm_i_math_thunder:
+
+Inference methods: *smo* and *thunder*
+--------------------------------------
+*smo* and *thunder* inference methods performs prediction by the same way:
+
+Given the SVM classifier and :math:`r` feature vectors :math:`x_1, \ldots, x_r`, 
+the problem is to calculate the signed value of the
+decision function :math:`D(x_i)`, :math:`i=1, \ldots, r`. The sign of the
+value defines the class of the feature vector, and the absolute
+value of the function is a multiple of the distance between the
+feature vector and separating hyperplane.
+
+---------------------
+Programming Interface
+---------------------
+All types and functions in this section shall be declared in the
+``oneapi::dal::svm`` namespace and be available via inclusion of the
+``oneapi/dal/algo/svm.hpp`` header file.
+                   
+Descriptor
+----------
+.. onedal_class:: oneapi::dal::svm::detail::v1::descriptor_base
+.. onedal_class:: oneapi::dal::svm::v1::descriptor
+
+Method tags
+~~~~~~~~~~~
+.. onedal_tags_namespace:: oneapi::dal::svm::method::v1
+
+Task tags
+~~~~~~~~~
+.. onedal_tags_namespace:: oneapi::dal::svm::task::v1
+
+Model
+-----
+.. onedal_class:: oneapi::dal::svm::v1::model
+
+.. _svm_t_api:
+
+Training :expr:`train(...)`
+--------------------------------
+.. _svm_t_api_input:
+
+Input
+~~~~~
+.. onedal_class:: oneapi::dal::svm::v1::train_input
+
+
+.. _svm_t_api_result:
+
+Result
+~~~~~~
+.. onedal_class:: oneapi::dal::svm::v1::train_result
+
+Operation
+~~~~~~~~~
+.. onedal_func:: oneapi::dal::v1::train
+
+.. _svm_i_api:
+
+Inference :expr:`infer(...)`
+----------------------------
+.. _svm_i_api_input:
+
+Input
+~~~~~
+.. onedal_class:: oneapi::dal::svm::v1::infer_input
+
+
+.. _svm_i_api_result:
+
+Result
+~~~~~~
+.. onedal_class:: oneapi::dal::svm::v1::infer_result
+
+Operation
+~~~~~~~~~
+.. onedal_func:: oneapi::dal::v1::infer
```

Comment:
Operations should be included like this (same for train):

![image](https://user-images.githubusercontent.com/17865025/101584486-d5fc1800-39ee-11eb-8fe4-49a039a0c50f.png)